util/ranger/detacher.go

```diff
@@ -384,12 +486,31 @@ func ExtractEqAndInCondition(sctx sessionctx.Context, conditions []expression.Ex
 	for i, ma := range mergedAccesses {
 		if ma == nil {
 			if accesses[i] != nil {
-				newConditions = append(newConditions, accesses[i])
+				if allEqOrIn(accesses[i]) {
+					newConditions = append(newConditions, accesses[i])
+				} else {
+					accesses[i] = nil
+				}
 			}
 			continue
 		}
-		accesses[i] = points2EqOrInCond(sctx, points[i], cols[i])
-		newConditions = append(newConditions, accesses[i])
+		points[i] = allSinglePoints(sctx.GetSessionVars().StmtCtx, points[i])
+		if points[i] == nil {
+			// There exists an interval whose length is larger than
```

Comment:
larger than what?

Reply:
Larger than 0. Sorry that I forgot to complete the comment.